app/models/dimensions/date.rb

```diff
@@ -30,12 +30,8 @@ def self.create_with(date)
     date_dimension
   end
 
-  def self.find_or_create(date)
-    begin
-      find_by(date: date) || create_with(date)
-    rescue ActiveRecord::RecordNotUnique
```

Comment:
We still have race conditions where concurrent executions both "find" at the same time, and then attempt to create.